crates/ty_python_semantic/src/types/constraints.rs

```diff
@@ -2279,6 +2304,18 @@ impl<'db> SequentMap<'db> {
         left_constraint: ConstrainedTypeVar<'db>,
         right_constraint: ConstrainedTypeVar<'db>,
     ) {
+        // These might seem redundant with the intersection check below, since `a → b` means that
+        // `a ∧ b = a`. But we are not normalizing constraint bounds, and these clauses help us
+        // identity constraints that are identical besides e.g. ordering of union/intersection
+        // elements. (For instance, these clauses would find `T ≤ τ₁ ∩ τ₂` and `T ≤ τ₂ ∩ τ₁`, and
+        // add sequents for `(T ≤ τ₁ ∩ τ₂) → (T ≤ τ₂ ∩ τ₁)` and vice versa.)
+        if left_constraint.implies(db, right_constraint) {
+            self.add_single_implication(left_constraint, right_constraint);
+        }
+        if right_constraint.implies(db, left_constraint) {
+            self.add_single_implication(right_constraint, left_constraint);
+        }
```

Comment:
The explanation above seems to imply that we could (should?) return early here? Because if `a → b` always implies `a ∧ b = a`, then the code below would otherwise add *two additional* `a → b` implications and one `a → a` implication, both of which seem unnecessary?